public/pages/AnomalyCharts/components/FeatureChart/FeatureChart.tsx

```diff
@@ -187,7 +203,12 @@ export const FeatureChart = (props: FeatureChartProps) => {
           opacity: showLoader ? 0.2 : 1,
         }}
       >
-        <Chart>
+        {/**
+         * Charts may show stale data even after feature data is changed.
```

Comment:
You mean even `props.featureData` changed the chart won't refresh itself? Looks like some bug of `Chart`?

Reply:
Tried debugging for awhile, could either be a bug in EUI's `Chart` or because of the fact that `featureData` is an array, and there's some caveats where arrays as state variables won't trigger an update, if the values are simply changed (see https://stackoverflow.com/questions/56266575/why-is-usestate-not-triggering-re-render). I've tried some of the suggestions there with no luck, so I'm guessing some issue with `Chart`.

I was able to see the list get updated and log outputs that it is updated, but the chart remains the same. This fix just guarantees a re-render.

Reply:
Got it. If set the key as `featureData`, how will the rendered HTML look like? The id will be the the whole content of `featureData` ? I'm ok with this workaround, just curious if it will generate long id.

Reply:
Interesting question - I checked and don't see it getting rendered in HTML at all. I believe that's because `key` is a react-specific prop for allowing user-controlled re-rendering, and isn't necessary to include in the produced HTML. `id` is an HTML-specific attribute, which would get rendered in HTML.

Reply:
Got it, that should be ok